packages/core/src/App/Containers/Routes/routes.jsx

```diff
@@ -34,21 +45,71 @@ class Routes extends React.Component {
         this.props.setAppRouterHistory(this.props.history);
     }
 
+    componentDidUpdate(pp, ps) {
+        if (pp.location.pathname !== this.props.location.pathname) {
+            this.isBot(this.props.location);
+        }
+        if (this.state.is_bot !== ps.is_bot) {
+            if (this.state.is_bot) {
+                this.removeBlockerEvents();
+                this.el_landscape_blocker.classList.add('landscape-blocker--keyboard-visible');
+            } else {
+                this.addBlockerEvents();
+                this.el_landscape_blocker.classList.remove('landscape-blocker--keyboard-visible');
+            }
+        }
+    }
+
     componentWillUnmount() {
+        this.removeBlockerEvents();
         if (typeof this.unlisten_to_change === 'function') {
             this.unlisten_to_change();
             this.unlisten_to_change = null;
             this.initial_route = null;
         }
     }
 
+    isBot = location => {
```

Comment:
- We have an `isBot` helper that you can use in `deriv-app/packages/shared/src/utils/platform/platform.js`
    - If you're using above function, you could get rid of this function altogether and just update the state in `componentDidUpdate`.
- Since this sets a value rather than returning a value a name like `setIsBot` would be more appropriate.

Reply:
Aaron I've checked it and I think if I use isBot() function from share, we have more latency to remove blocker when we back from a blocked page to dbot

Reply:
Why would there be more latency? 🤔